File: 0129-sum-root-to-leaf-numbers/0129-sum-root-to-leaf-numbers.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */

func dfs (current *TreeNode , num int ) int{
    if current == nil {
        return 0 
    }
    
    num = num * 10 + current.Val
    
    if current.Left == nil && current.Right == nil {
        return num 
    }
    
    return dfs(current.Left , num ) + dfs(current.Right , num )
}

func sumNumbers(root *TreeNode) int {
    return dfs(root , 0)
    
}

File: 0129-sum-root-to-leaf-numbers/0129-sum-root-to-leaf-numbers.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */

function dfs(curr, num) {
    if(!curr) return 0  
    
    num = num * 10  + curr.val
    
    if( !curr.left && !curr.right ) return num 
    return dfs(curr.left , num ) + dfs(curr.right , num)
}

var sumNumbers = function(root) {
    
        return dfs(root , 0)
    
};